src/Raven.Server/Documents/Handlers/StreamCsvDocumentQueryResultWriter.cs

```diff
@@ -101,9 +102,13 @@ private void WriteCsvHeaderIfNeeded(Document res)
             _csvWriter.NextRecord();
         }
 
-        private IEnumerable<(string Property, string Path)> GetPropertiesRecursive((string ParentProperty, string ParentPath) propertyTuple, BlittableJsonReaderObject obj)
+        private IEnumerable<(string Property, string Path)> GetPropertiesRecursive((string ParentProperty, string ParentPath) propertyTuple, BlittableJsonReaderObject obj, bool addId = true)
         {
             var inMetadata = Constants.Documents.Metadata.Key.Equals(propertyTuple.ParentPath);
+            if (addId)
```

Comment:
How do we prevent from adding `@id` column twice when doing export csv in collection? (according to issue description `@id` was already exported then)

Reply:
This code is only invoked when not specifying the fields and it is taken from the first document. When specifying fields the fields you specified will be in the header.

Reply:
OK